### PR TITLE
Feat: Update `rhino-showcase` to Rhino 1.2.0

### LINKED ISSUE #17: Upgrade Rhino Showcase to the latest Rhino version

### PR DESCRIPTION
**Pull request by vibalre**
### Changes
Closes #17 

### How to test
* Update to Rhino 1.2.0.
* Clone [chore/update-rhino-version](https://github.com/Appsilon/rhino-showcase/tree/chore/update-rhino-version).
* Run `rhino:app()`.

### Notes
* There is a large difference in `renv.lock` and `renv` directory because I updated the dependencies to the latest versions.
* The app has been updated to Rhino 1.2.0, but there is an issue with the `plotly` and some `shiny` elements no rendering properly.

### CODE REVIEW

**Comment by jakubnowicki** [comment]
Why Rhino showcase has still the old (shiny-template style) structure?

**Comment by vibalre** [comment]
@jakubnowicki When you say old structure, do you mean the use of `app/r` instead of `app/logic` and `app/view`? If that is the case, @kamilzyla advised to stick with `app/r` for now.

**Comment by jakubnowicki** [comment]
@kamilzyla what is the reason not to move to the new structure right now? I believe implementing that is not a complicated task and this would unblock further refactoring of the application (even using bench time), so it can serve as an actual showcase.

**Comment by kamilzyla** [comment]
@jakubnowicki Changing the structure will require quite a bit of work: the code was not written with logic/view separation in mind, so it's not as simple as moving some files around. This PR is already quite complex, so it's better to handle it in a separate task: #19.

**Review by kamilzyla — CHANGES_REQUESTED** [review]
Looking good overall, great job! :tada:

I allowed myself to push a some commits to your branch:
1. Remove (duplicate and unused) PR template from `.github/workflows` (it's already been there when we started the upgrade, just fixing it along the way).
2. Remove mention of Yarn from README.
3. Add `.rscignore` - you might have forgotten to copy it over.
4. Rebuild `app/js/app.min.js`. Even though the JS sources didn't change, we made some updates to the build pipeline since Rhino 0.4.0, so the minified file is slightly different too.
5. Rename `src.Rproj` to `rhino-showcase.Rproj` and update it.

Otherwise everything seems to be working fine, but... the plots are not showing up and we have an exception in the JS console of the running application (`Uncaught TypeError`). We need to fix it before merging. I didn't have the time to investigate it; LMK if you need any help.

**Comment by vibalre** [comment]
I found the cause of the plots not rendering properly. It is because of the breaking changes in JavaScript API in [`plotly` 4.10.0](https://github.com/plotly/plotly.R/releases/tag/v4.10.0).
I reverted to `plotly` 4.9.4.1.
```
renv::install("plotly@4.9.4.1")
renv::snapsnot()
renv::restore()
```
Plots are working now after the reverting `plotly`.
I still need to investigate what needs to change in the code with regards to `plotly`. Quick solution would be to revert `plotly` in `renv.lock`, but I think it will be better if we can make `rhino-showcase` works with latest `plotly` version.

**Comment by vibalre** [comment]
@kamilzyla Reverted `plotly` to 4.9.4.1. Plots should render properly now.

File: renv.lock
Comment (kamilzyla) [inline renv.lock:1004]: Please update to 1.2.1.

**Review by kamilzyla — APPROVED** [review]
LGTM :+1: Please upgrade Rhino to 1.2.1 (install from CRAN) before merging.

**Pull request merged**
